var/spack/repos/builtin/packages/elemental/package.py

```diff
@@ -0,0 +1,131 @@
+    # Note that this forces us to use OpenBLAS until #1712 is fixed
+    depends_on('lapack', when='~openmp_blas')
+    depends_on('metis')
+    depends_on('metis +int64', when='+int64')
+    depends_on('mpi')
+    # Allow Elemental to build internally when using 8-byte ints
+    depends_on('scalapack', when='+scalapack ~int64_blas')
+    depends_on('python', when='+python')
+
+    def build_type(self):
+        """Returns the correct value for the ``CMAKE_BUILD_TYPE`` variable
+        :return: value for ``CMAKE_BUILD_TYPE``
+        """
+        if '+debug' in self.spec:
+            return 'Debug'
+        else:
+            return 'Release'
+
+    def cmake_args(self):
+        args = ['-DCMAKE_INSTALL_MESSAGE:STRING=LAZY',
+                '-DEL_PREFER_OPENBLAS:BOOL=TRUE',
+                '-DEL_DISABLE_SCALAPACK:BOOL={0}'.format((
+                    'OFF' if '+scalapack' in self.spec else 'ON')),
+                '-DGFORTRAN_LIB=libgfortran.so',
+                '-DBUILD_SHARED_LIBS:BOOL={0}'.format((
+                    'ON' if '+shared' in self.spec else 'OFF')),
+                '-DEL_HYBRID:BOOL={0}'.format((
+                    'ON' if '+hybrid' in self.spec else 'OFF')),
+                '-DEL_C_INTERFACE:BOOL={0}'.format((
+                    'ON' if '+c' in self.spec else 'OFF')),
+                '-DINSTALL_PYTHON_PACKAGE:BOOL={0}'.format((
+                    'ON' if '+python' in self.spec else 'OFF')),
+                '-DEL_DISABLE_PARMETIS:BOOL={0}'.format((
+                    'OFF' if '+parmetis' in self.spec else 'ON')),
+                '-DEL_DISABLE_QUAD:BOOL={0}'.format((
+                    'OFF' if '+quad' in self.spec else 'ON')),
+                '-DEL_USE_64BIT_INTS:BOOL={0}'.format((
+                    'ON' if '+int64' in self.spec else 'OFF')),
+                '-DEL_USE_64BIT_BLAS_INTS:BOOL={0}'.format((
+                    'ON' if '+int64_blas' in self.spec else 'OFF'))]
+        if '+int64_blas' in self.spec:
+            args.extend(['-DEL_BLAS_SUFFIX:STRING={0}'.format((
+                '_64_' if '+int64_blas' in self.spec else '_')),
+                         '-DCUSTOM_BLAS_SUFFIX:BOOL=TRUE']),
+            if '+scalapack' in self.spec:
+                args.extend(['-DEL_LAPACK_SUFFIX:STRING={0}'.format((
+                    '_64_' if '+int64_blas' in self.spec else '_')),
+                             '-DCUSTOM_LAPACK_SUFFIX:BOOL=TRUE']),
+        else:
+            lapack = self.spec['lapack'].lapack_libs
+            blas = self.spec['blas'].blas_libs
+
+            if '+scalapack' in self.spec:
+                scalapack = self.spec['scalapack'].scalapack_libs
+                args.extend(['-DMATH_LIBS:STRING={0} {1}'.format(
+                    (lapack + blas).search_flags, scalapack.search_flags),
+                             '-DMATH_LIBS:STRING={0} {1}'.format(
+                    (lapack + blas).ld_flags.split()[1], scalapack.ld_flags.split()[1])])
```

Comment:
Are you getting `-lfoo -lbar -lbaz` here? If so, there is `link_flags()`. Note that `ld_flags` may have more than one directory set with `-L`.
(same below)

Also, should the order be the opposite (scalapack first)?